### PR TITLE
Promote error to Revert even if decoding the reason fails

### PR DESCRIPTION
**Pull request by sunce86**
Return `Revert` status even if decoding the reason failed.

Should fix the error https://cowservices.slack.com/archives/C0371SB243E/p1665417220790089

This error appears when `eth_call` batch request is sent for `TokenOwnerFinder`: https://github.com/cowprotocol/services/blob/7f2abc427f8627bc712b54ae61977f43c3159a16/crates/shared/src/bad_token/token_owner_finder.rs#L231-L233

Nethermind node asnwers with error message `"Reverted 0x416d6f756e74206d757374206265206c657373207468616e20746f74616c207265666c656374696f6e73"` for one of the requests, which is detected as reverted, but fails to decode `416d6f756e74206d757374206265206c657373207468616e20746f74616c207265666c656374696f6e73` to a reason. This causes this error not to be promoted to `ExecutionError::Revert` but instead to `ExecutionError::Web3` which causes `TokenOwnerFinder` to yield error.

Note: I've spent quite some time to debug why the decoding of the message is failing, but failed to figure out. Tried also to investigate latest changes in Nethermind but with no success. If anyone else is willing to take a look, that would be awesome.

### CODE REVIEW

**Comment by vkgnosis** [comment]
You are changing the implementation of the parity error detector but the we are trying to decode a nethermind error. I don't think the error format is standardized in any way so it is not surprising that the parity detector cannot handle it and changing this detector seems like the wrong way to go.

Maybe a better is to change https://github.com/gnosis/ethcontract-rs/blob/05bff3fb254dbd0cf6b198b3d4a547d9403acdf4/ethcontract/src/errors.rs#L133 to also try a new `nethermind::get_encoded_error` which could decode this way. We can start with that.
Then it would be still be nice to extract the encoded revert reason for which we'd have to figure out how Nethermind encodes these bytes. We could ask the Nethermind developers about that.

**Review by vkgnosis — APPROVED** [review]
This is ok to me now but would wait for Nic's review before merging. One thing I am still wondering is if we have ever successfully decoded a revert reason from nethermind. Maybe the decode_reason function just doesn't apply to it at all.

**Comment by sunce86** [comment]
Maybe we never used Nethermind node with the `tracing` feature. In that case we would get `Reverted 0x` every time and that would never enter the decoding part.

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
Looks like the CI is failing because of a Rinkeby example test. Feel free to disable it for now (you can just comment them out).

https://github.com/gnosis/ethcontract-rs/blob/05bff3fb254dbd0cf6b198b3d4a547d9403acdf4/.github/workflows/pull-request.yaml#L64-L68

File: ethcontract/src/errors/nethermind.rs
Comment (nlordell) [inline ethcontract/src/errors/nethermind.rs:36]: We can also check:

```rust
if err.message == "VM execution error" {
    return Some(ExecutionError::Revert(None));
}
```

As a catch-all (in case we missed some other error messages - like what happens when a contract reverts without data?).

Comment (sunce86) [reply ethcontract/src/errors/nethermind.rs:36]: I don't see a reason why not to add it. Done.

**Comment by sunce86** [comment]
> Looks like the CI is failing because of a Rinkeby example test. Feel free to disable it for now (you can just comment them out).

I've commented out the rinkeby line but nothing changed. 
Merging since the CI errors don't seem too frightening :)

**Pull request merged**
